jdbi/src/main/java/org/killbill/commons/jdbi/guice/DataSourceProvider.java

```diff
@@ -230,6 +243,57 @@ DataSource buildDataSource() {
         }
     }
 
+    private class UnpooledDataSourceBuilder {
+
+        DataSource buildDataSource() {
+            String dataSourceClassName = config.getDataSourceClassName();
+            if (Strings.isNullOrEmpty(dataSourceClassName)) {
+                throw new IllegalArgumentException("dataSourceClassName was not specified in the configuration and could not be inferred");
+            }
+
+            if (DatabaseType.MYSQL.equals(databaseType)) {
+                Reference ref = new Reference(dataSourceClassName);
+                ref.add(new StringRefAddr("user", config.getUsername()));
+                ref.add(new StringRefAddr("password", config.getPassword()));
+                ref.add(new StringRefAddr("explicitUrl", "true"));
+                ref.add(new StringRefAddr("url", config.getJdbcUrl()));
+                try {
+                    return (DataSource) new MysqlDataSourceFactory().getObjectInstance(ref, null, null, null);
```

Comment:
Have you looked at re-using parts of HikariCP instead? Something like a custom `UnpooledDataSource` which calls `ProxyFactory.getProxyConnection` (see `BaseHikariPool`) to create connections?

I agree that it's a bit odd to use part of the pooling code to implement an unpooled DataSource but among advantages:

* it would prevent us from having to hardcode all the per-driver properties (e.g. it seems that `serverName` is missing for MySQL)
* no need to pull all drivers in killbill-jdbi
* no need for MariaDB vs MySQL custom configuration (we would return `com.mysql.jdbc.jdbc2.optional.MysqlDataSource` here but we actually use `org.mariadb.jdbc.MySQLDataSource` by default today)